### PR TITLE
Update OTel SDK to 1.17

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PlainTextAllocationEventExporter.java [hunk 1]
```diff
@@ -83,17 +81,14 @@ public void export(RecordedEvent event, AllocationEventSampler sampler, SpanCont
...
-        LogDataBuilder.create(resource, INSTRUMENTATION_SCOPE_INFO)
-            .setEpoch(time)
-            .setBody(body)
-            .setAttributes(attributes);
+    LogRecordBuilder logRecordBuilder =
+        logEmitter.logRecordBuilder().setEpoch(time).setBody(body).setAllAttributes(attributes);
 
     if (spanContext != null && spanContext.isValid()) {
-      logDataBuilder.setSpanContext(spanContext);
+      logRecordBuilder.setContext(Context.root().with(Span.wrap(spanContext)));
```
Comment (breedx-splk) [inline profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PlainTextAllocationEventExporter.java:88]: What's up with this? Does this create a new context with a new span based on the existing context? I know you didn't write it, just trying to understand what/why...

File: profiler/src/test/java/com/splunk/opentelemetry/profiler/TLABProcessorTest.java [hunk 2]
```diff
@@ -152,14 +164,20 @@ private void tlabProcessTest(Long tlabSize) {
...
+                    .hasSpanContext(spanContext)
+                    .hasBody(stackAsString)
+                    .hasEpochNanos(
+                        TimeUnit.SECONDS.toNanos(now.getEpochSecond()) + clock.nanoTime())
+                    .hasAttributes(
+                        entry(SOURCE_TYPE, "otel.profiling"),
+                        entry(SOURCE_EVENT_NAME, "tee-lab"),
+                        entry(ALLOCATION_SIZE_KEY, ONE_MB),
+                        entry(DATA_TYPE, ProfilingDataType.ALLOCATION.value()),
+                        entry(DATA_FORMAT, Configuration.DataFormat.TEXT.value())));
```
Comment (breedx-splk) [inline profiler/src/test/java/com/splunk/opentelemetry/profiler/TLABProcessorTest.java:180]: Nice. 👍🏻

**Review by breedx-splk — APPROVED** [review]
Considerably larger than I had expected, heh. Progress!
I did have one question, but looks good overall...thanks for jumping on this.

**Comment by breedx-splk** [comment]
Looks like `SplunkMetricsConfigTest` also needs a little help.

**Comment by mateuszrzeszutek** [comment]
> Looks like `SplunkMetricsConfigTest` also needs a little help.

Hah, weird, I fixed it yesterday but forgot to commit&push 🙈

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PlainTextAllocationEventExporter.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PlainTextAllocationEventExporter.java:88]: Yeah, pretty much. The `Span.wrap()` function creates just a shell/wrapper, not a fully fledged span though.

Unfortunately you can't just pass a `SpanContext` to a `LogRecordBuilder` directly, it has to be a `Context`.

**Review by laurit — APPROVED** [review]

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PlainTextAllocationEventExporter.java (hunk 1 above)
Comment (breedx-splk) [reply profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PlainTextAllocationEventExporter.java:88]: Ah, cool, thanks for clarifying `Context` vs. `SpanContext` (wasn't immediately obv to me, sorry). 👍🏻

**Pull request merged**